modules/auxiliary/scanner/http/git_scanner.rb

```diff
@@ -0,0 +1,126 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::WmapScanServer
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name'        => 'HTTP Git Scanner',
+      'Description' => 'Detect git directories and files and analize its content.',
+      'Author'       => ['t0nyhj'],
+      'License'     => MSF_LICENSE
+    )
+
+    register_options(
+      [
+        OptString.new('PATH', [ true,  "The test path to .git directory", '/'])
+
+      ], self.class)
+
+    register_advanced_options(
+      [
+        OptInt.new('ErrorCode', [ true, "Error code for non existent directory", 404]),
+        OptPath.new('HTTP404Sigs',   [ false, "Path of 404 signatures to use",
+            File.join(Msf::Config.data_directory, "wmap", "wmap_404s.txt")
+          ]
+        )
+
+      ], self.class)
+  end
+
+  def run_host(target_host)
+    conn = true
+    ecode = nil
+    emesg = nil
+
+    tpath = normalize_uri(datastore['PATH'])
+
+    ecode = datastore['ErrorCode']
+    vhost = datastore['VHOST'] || wmap_target_host
+
+    #
+    # Detect error code
+    #
+    begin
+      randdir = Rex::Text.rand_text_alpha(5).chomp + '/'
```

Comment:
Please use snake_case for symbols, methods and variables. https://github.com/bbatsov/ruby-style-guide#snake-case-symbols-methods-vars

It applies to the full module.